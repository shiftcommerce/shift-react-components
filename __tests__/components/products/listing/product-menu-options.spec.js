// Libraries
import React from 'react'

// Component
import ProductMenuOptions from '../../../../src/components/products/listing/product-menu-options'

// Objects
import Breadcrumb from '../../../../src/objects/breadcrumb'

test('renders correctly', () => {
  // Arrange
  const initialProps = {
    toggleFiltering: () => (null),
    filterCount: 3,
    indexName: 'test_index'
  }

  // Act
  const wrapper = shallow(
    <ProductMenuOptions {...initialProps} />
  )

  const rootTree = wrapper.find('.c-product-listing__menu-options')

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(rootTree.find(Breadcrumb))
})
