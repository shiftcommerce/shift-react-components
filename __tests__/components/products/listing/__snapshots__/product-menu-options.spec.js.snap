// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="c-product-listing__menu-options"
>
  <div
    className="c-product-listing__menu-options-filters"
  >
    <h2
      className="c-product-listing__menu-options-filters-title"
    >
      Filters
    </h2>
    <div
      className="c-product-listing__menu-options-filters-applied"
    >
      <AlgoliaCurrentRefinements(UnknownComponent) />
    </div>
    <Button
      className="c-product-listing__menu-options-filters-button"
      onClick={[Function]}
    />
  </div>
  <div
    className="c-product-listing__menu-options-sort-by"
  >
    <h2
      className="c-product-listing__menu-options-sort-by-title"
    >
      Sort by:
    </h2>
    <AlgoliaSortBy(SortBy)
      defaultRefinement="test_index"
      items={
        Array [
          Object {
            "label": "Featured",
            "value": "test_index",
          },
          Object {
            "label": "Rating asc.",
            "value": "test_index_product_rating_asc",
          },
          Object {
            "label": "Rating desc.",
            "value": "test_index_product_rating_desc",
          },
        ]
      }
    />
    <Button
      className="c-product-listing__menu-options-sort-by-button u-hidden-d"
    />
  </div>
</div>
`;
